src/geometric_utils.ts

```diff
@@ -126,17 +214,178 @@ export class GeometricUtils {
+                if (poly1[i][0] === poly2[j][0] && poly1[i][1] === poly2[j][1]) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    // Check if two polygons share an edge, ie if they contain an identical line segment
+    public static polygons_share_edge(poly1: ULabelSpatialPayload2D, poly2: ULabelSpatialPayload2D): boolean {
+        for (let i: number = 0; i < poly1.length-1; i++) {
+            let line1: LineSegment2D = [poly1[i], poly1[i+1]];
+            // skip if the points are the same
+            if (GeometricUtils.points_are_equal(line1[0], line1[1])) {
+                continue;
+            }
+            for (let j: number = 0; j < poly2.length-1; j++) {
+                let line2: LineSegment2D = [poly2[j], poly2[j+1]];
+                // skip if the points are the same
+                if (GeometricUtils.points_are_equal(line2[0], line2[1])) {
+                    continue;
+                }
+                if (GeometricUtils.line_segments_are_on_same_line(line1, line2)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+
+    }
+
+    // Scale a polygon about a center point, or the centroid if no center is provided
+    public static scale_polygon(poly: ULabelSpatialPayload2D, scale, center: Point2D = null): ULabelSpatialPayload2D {
+        let ret: ULabelSpatialPayload2D = [];
+        if (center === null) {
+            // use the centroid
+            center = GeometricUtils.get_centroid_of_polygon(poly);
+        }
+
+        for (let i: number = 0; i < poly.length; i++) {
+            let pt: Point2D = poly[i];
+            let new_x: number = center[0] + (pt[0] - center[0])*scale;
+            let new_y: number = center[1] + (pt[1] - center[1])*scale;
+            ret.push([new_x, new_y]);
+        }
+        return ret;
+    }
+
+    // Get the centroid of a polygon
+    public static get_centroid_of_polygon(poly: ULabelSpatialPayload2D): Point2D {
+        let x: number = 0;
+        let y: number = 0;
+        for (let i: number = 0; i < poly.length; i++) {
+            x += poly[i][0];
+            y += poly[i][1];
+        }
+        return [x/poly.length, y/poly.length];
+    }
+
+    // Check if poly1 is completely within poly2 by checking if any line segment of poly1 intersects with poly2
+    public static polygon_is_within_polygon(poly1: ULabelSpatialPayload2D, poly2: ULabelSpatialPayload2D): boolean {
+        if (GeometricUtils.is_polygon_closed(poly1) && GeometricUtils.is_polygon_closed(poly2)) {
+            return turf.booleanWithin(turf.polygon([poly1]), turf.polygon([poly2]));
+        } else {
+            return false;
+        }
+    }
+
+    // Check if a point is within a polygon
+    public static point_is_within_polygon(point: Point2D, poly: ULabelSpatialPayload2D): boolean {
```

Comment:
Another candidate for https://turfjs.org/docs/#booleanWithin?

Reply:
yup, this one I changed in #132, see [d886be0](https://github.com/SenteraLLC/ulabel/pull/132/commits/d886be019fe15946981d59b381e3dfe3d5416563)